smart_contract/src/karma_sc_test.rs

```diff
@@ -0,0 +1,411 @@
+            .network::<Ethereum>()
+            .connect_ws(ws_connect)
+            .await?;
+
+        Ok(KarmaSC::new(address, provider))
+    }
+}
+
+impl KarmaTiersSC::KarmaTiersSCInstance<AlloyWsProvider> {
+    pub async fn try_new(
+        rpc_url: Url,
+        address: Address,
+    ) -> Result<Self, Box<dyn std::error::Error + Send + Sync>> {
+        let ws_connect = WsConnect::new(rpc_url.as_str());
+
+        let provider = ProviderBuilder::new()
+            .network::<Ethereum>()
+            .connect_ws(ws_connect)
+            .await?;
+
+        Ok(KarmaTiersSC::new(address, provider))
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
+    println!("Testing Karma Contract Interaction with Signed Types");
+
+    let rpc_url = "wss://public.sepolia.rpc.status.network/ws";
+    let karma_contract_address = "0x9ADD5A2F68d0d95F74C751a2081EFF57Ff1C836F";
+    let karma_tiers_contract_address = "0x011b9de308BE357BbF24EfB387a270a14A04E5d2";
+    let test_account = "0x360a45F70De193090A1b13dA8393A02F9119aeCd";
+
+    println!("Connecting to RPC: {}", rpc_url);
+
+    let karma_contract_addr = Address::from_str(karma_contract_address)
+        .map_err(|e| format!("Invalid karma contract address: {}", e))?;
+    let karma_tiers_contract_addr = Address::from_str(karma_tiers_contract_address)
+        .map_err(|e| format!("Invalid karma tiers contract address: {}", e))?;
+    let account_addr =
+        Address::from_str(test_account).map_err(|e| format!("Invalid account address: {}", e))?;
+
+    let url = Url::parse(rpc_url)?;
+
+    // Connect to Karma contract
+    let karma_contract =
+        match KarmaSC::KarmaSCInstance::try_new(url.clone(), karma_contract_addr).await {
+            Ok(contract) => {
+                println!(
+                    "Successfully connected to Karma contract at {}",
+                    karma_contract_addr
+                );
+                contract
+            }
+            Err(e) => {
+                eprintln!("Failed to connect to karma contract: {}", e);
+                return Err(e);
+            }
+        };
+
+    // Connect to KarmaTiers contract
+    let karma_tiers_contract =
+        match KarmaTiersSC::KarmaTiersSCInstance::try_new(url, karma_tiers_contract_addr).await {
+            Ok(contract) => {
+                println!(
+                    "Successfully connected to KarmaTiers contract at {}",
+                    karma_tiers_contract_addr
+                );
+                contract
+            }
+            Err(e) => {
+                eprintln!("Failed to connect to karma tiers contract: {}", e);
+                return Err(e);
+            }
+        };
+
+    println!("\nTesting Contract Functions:");
+    println!("=====================================");
+
+    // Test Karma Contract Functions with signed types
+    println!("\n--- Karma Contract Tests ---");
+
+    // Test 1: Get account balance
```

Comment:
Should only test karma_amount here, everething else (e.g. not used in prover) should have been tested by the SC team